geode-gfsh/src/main/java/org/apache/geode/management/internal/cli/commands/StopGatewaySenderCommand.java

```diff
@@ -36,7 +44,30 @@
 import org.apache.geode.security.ResourcePermission;
 
 public class StopGatewaySenderCommand extends GfshCommand {
+  private final ExecutorService executorService;
+  private SystemManagementService managementService;
+  private BiFunction<String[], String[], Set<DistributedMember>> findMembers;
+  private final Supplier<StopGatewaySenderOnMember> stopperOnMemberFactory;
+
+  @SuppressWarnings("unused") // invoked by spring shell
+  public StopGatewaySenderCommand() {
+    this(newCachedThreadPool("Stop Sender Command Thread ", true),
+        StopGatewaySenderOnMemberWithBeanImpl::new, null, null);
```

Comment:
`GfshCommand` has `findMember()` and `getManagementService()` you can stubbed out in your tests, I don't think you need the last two params in your "test constructor". Also, for the first two parameters, one you are passing the object itself, one is a supplier, I suggest you make it consistent, I would pass in both in object form (change the supplier to be an instance of `StopGatewaySenderOnMember`.

Reply:
I am passing the `SystemManagmentService` instance and the `findMembers()` function to avoid using a spy, which is not considered a good practice unless you are forced to because you are adding tests to legacy code.
I do not know in this case, given that the methods are provided by the base class, using a spy would be justified.

I have changed the code to pass to the constructor an instance of `StopGatewaySenderOnMember` instead of the supplier as you have suggested.

Reply:
"Spying an object you are testing has been communicated to me as bad practice, but these warnings come with little to no explanation backing it up. I'm sure, like anything else, it can definitely be abused.". 
To me, in this case, spying on these methods to pass in a mock object is definitely better than passing null objects in the product calling code.

Reply:
This is probably one of those cases in which using a spy is justified. I cannot give you strong arguments in this case that make the previous implementation without spy better, so I will follow your proposal.